src/index.js

```diff
@@ -8,11 +8,23 @@ const fs_stat = promisify(fs.stat);
 
 async function search(dir, regex, depth, result = [], concurrency) {
     async function fileAnalyzer(file) {
-        const stat = await fs_stat(path.join(dir, file));
-        if (stat.isFile() && regex.test(file)) {
-            result.push({ dir, file });
-        } else if (stat.isDirectory() && depth > 0) {
-            await search(path.join(dir, file), regex, depth - 1, result, concurrency);
+        const statPath = path.join(dir, file);
+        const stat = await fs_stat(statPath);
+
+        if (!regex.global) {
+            if (stat.isFile() && regex.test(file)) {
+                result.push({ dir, file });
+            } else if (stat.isDirectory() && depth > 0) {
+                await search(path.join(dir, file), regex, depth - 1, result, concurrency);
+            }
+        } else { // scan the entire path for the regex if the pattern uses a //g
+            if (regex.test(statPath)) {
+                result.push({ dir, file });
+            }
+
+            if (stat.isDirectory() && depth > 0) {
```

Comment:
*Please try to avoid code duplication*
Duplicate code with reference to lines 17-19.

Reply:
Fixed. Refactored in latest commit.

Reply:
Unit tests are also ready for review. I copied what you had and did the same for the second codepath (//g). I also added a new test that tests the ability to search in the filepath itself.